notifications_utils/clients/redis/redis_client.py

```diff
@@ -130,6 +129,64 @@ def exceeded_rate_limit(self, cache_key, limit, interval, raise_exception=False)
         else:
             return False
 
+    def add_key_to_sorted_set(self, cache_key, value, score, raise_exception=False):
+        """
+        Add a key to a sorted set, with a score
+        :param cache_key:
+        :param value:
+        :param score:
+        :param raise_exception:
+        """
+        cache_key = prepare_value(cache_key)
+        value = prepare_value(value)
+        if self.active:
+            try:
+                self.redis_store.zadd(cache_key, {value: score})
+            except Exception as e:
+                self.__handle_exception(e, raise_exception, "add-key-to-ordered-set", cache_key)
+
+    def delete_key_from_sorted_set(self, cache_key, min_score, max_score, raise_exception=False):
+        """
+        Delete a key from a sorted set
+
+        :param cache_key:
+        :param min_score:
+        :param max_score:
+        :param raise_exception:
+        """
+        cache_key = prepare_value(cache_key)
+        if self.active:
+            try:
+                self.redis_store.zremrangebyscore(cache_key, min_score, max_score)
+            except Exception as e:
+                self.__handle_exception(e, raise_exception, "delete-key-from-ordered-set", cache_key)
+
+    def get_length_of_sorted_set(self, cache_key, min_score=None, max_score=None, raise_exception=False):
```

Comment:
We don't want to follow previous pattern set by the `exceeded_rate_limit` function?

Reply:
so the exceed_rate_limit function was doing both the add to the sorted list + returning the length of the sorted list. We will need these to be different functions hence the split